src/vitest/describe-feature.ts

```diff
@@ -186,16 +186,16 @@ export function describeFeature (
     describe(feature.name, async () => {
         await featureFn(descibeFeatureParams)
 
-        beforeAll(() => {
-            beforeAllScenarioHook()
+        beforeAll(async () => {
+            await beforeAllScenarioHook()
         })
 
-        afterAll(() => {
+        afterAll(async () => {
             FeatureStateDetector
                 .forFeature(feature)
                 .checkNotCalledScenario()
             
-            afterAllScenarioHook()
+            await afterAllScenarioHook()
         })
 
         scenarioToRun.forEach((scenario) => scenario())
```

Comment:
I'm not sure about this line - do we also need to await the result of each run through `scenario()`?

I haven't tried making any local alterations to this, but my gut instinct tells me it should look something like this...

```suggestion
        for (const scenario of scenarioToRun) {
          await scenario();
        }
```

But this change might not be necessary (I'm not close enough to the code). And anyway, it's out of scope for this PR (whose purpose is just to make sure we `await` any asynchronous before/after logic).